src/theory/arith/constraint.h

```diff
@@ -896,10 +909,10 @@ class Constraint {
    */
   const ConstraintDatabase& getDatabase() const;
 
-private:
-
   /** Returns the constraint rule at the position. */
   const ConstraintRule& getConstraintRule() const;
+
+private:
```

Comment:
Can you fix the declaration order of this class according to `https://github.com/CVC4/CVC4/wiki/Code-Style-Guidelines#declaration-order`?

Reply:
Sure. A question about those guidelines.

Phrased abstractly: the guidelines describe two orderings: one by item kind (friend, type, constant, constructor-related, method, member) and another by visibility (public, protected, private). Which ordering is the macro-level order, and which is the micro-level ordering?

Concretely, would protected members or private methods come first?

Reply:
Perhaps 
>Do not alternate between sections.

means that the visibility order should be the macro-level ordering?

Reply:
Yes, visibility is macro-level ordering. The only exception is friend declarations (since there it doesn't matter if you declare them private or public), they should be on top. We'll clarify this in the code guidelines.

Reply:
Sounds good.  I've move the friend declaration to the top of `class Constraint`.